agentMET4FOF/metrological_streams.py

```diff
@@ -156,11 +163,20 @@ def __init__(
             quantity_units=quantity_units,
             misc=misc,
         )
+        self.value_unc = value_unc
+        self.time_unc = time_unc
         self.set_generator_function(
-            generator_function=self.sine_wave_function, sfreq=sfreq, F=F
+            generator_function=self._sine_wave_function,
+            uncertainty_generator=self._uncertainty_generator,
+            sfreq=sfreq,
+            F=F,
         )
 
-    def sine_wave_function(self, time, F=50):
+    def _sine_wave_function(self, time, F):
         """A simple sine wave generator"""
-        amplitude = np.sin(2 * np.pi * F * time)
+        amplitude = np.sin(2 * np.pi * F * time) + gauss(0, self.value_unc ** 2)
```

Comment:
Check the moving-average tutorial for another implementation and adapt it accordingly.

Reply:
moving average as in this one ?.. 

https://github.com/Met4FoF/agentMET4FOF/blob/develop/agentMET4FOF_tutorials/buffering/moving_average.py

Reply:
I was asking, because I thought you mentioned, that you implemented something similar to my white noise. In the linked tutorial you did apparently something different (which is aligned with what the name suggests). So, can we merge this here then or did I miss something?

Reply:
Yes, you're right, i thought i had it in the moving_average.py ,  i wanted to show the difference in the averaged signal , so the source signal had to be noisy.. i must have forgotten to commit